### ISSUE TITLE
Compilation without GStreamer

### ISSUE THREAD

**Issue by chazanov** [servo/servo#23050]
As GStreamer is a real hassle I would like to ask for the compilation option to compile without it.

**Comment by jdm** [comment]
This should be easy to accomplish in a few steps:
* update servo-media to a revision that includes https://github.com/servo/media/commit/a94a3b57ae5dbee9481fb98a1e2dd12e5303e8db (see ongoing work at #22944)
* add a Cargo feature to [the auto backend](https://github.com/servo/media/blob/master/backends/auto/Cargo.toml) which is used to enable the dummy backend
* add a Cargo feature to [the desktop app](https://github.com/servo/servo/blob/master/ports/servo/Cargo.toml) which plumbs the disable-gstreamer feature down to the crates which depend on servo-media-auto
* add a mach build argument to enable the disable-gstreamer feature

**Comment by chazanov** [comment]
@jdm Hi Josh, probably your time is precious.
May you anyway please give me some detailed guidance? I'm willing to learn Rust + Cargo, but currently I know nothing.

**Comment by paulrouget** [comment]
> This should be easy to accomplish in a few steps

I gave it a try. I'm not sure it's easy, we can't have a feature that "disable" something in Cargo.toml. The rustc feature will work, but something like this in Cargo.toml won't:

```toml
[target.'cfg(all(target="android", feature="disable-gstreamer"))'.dependencies]
```

It's a known limitation of Cargo (see https://github.com/rust-lang/cargo/issues/5499).

So I think we will need an intermediate crate (relevant irc conversation: https://mozilla.logbot.info/servo/20190514#c16308257).

I'm open to any other ideas.

**Comment by tuncer** [comment]
Fixed in #24668
